lib/net.c

```diff
@@ -1555,6 +1576,91 @@ start_network_stage2(void *arg)
 	return ret;
 }
 
+static int
+drop_privileges(const char *user)
+{
+	struct passwd *pw;
+
+	if ((pw = getpwnam(user)) == NULL) {
+		G_LOG(ERR, "%s: failed to call getpwnam() for user %s!\n",
+			__func__, user);
+		return -1;
+	}
+
+	/* Drop privileges. */
+	if (initgroups(user, pw->pw_gid) != 0 || setgid(pw->pw_gid) != 0 ||
+			setuid(pw->pw_uid) != 0) {
+		G_LOG(ERR, "%s: failed to drop privileges for user %s with user id %u and group id %u: %s!\n",
+			__func__, user, pw->pw_uid,
+			pw->pw_gid, strerror(errno));
+		return -1;
+	}
+
+	return (seteuid(0) == -1 && setegid(0) == -1) ? 0 : -1;
+}
+
+int
+change_file_permissions(const char *file_path,
+	const char *user, mode_t mode)
```

Comment:
Once `gatekeeper_log_init()` becomes responsible for changing the mode of the log file (see another comment for details on this), drop `mode` of the list of parameters of `change_file_permissions()`.